File: common/src/main/java/dev/skomlach/common/cryptostorage/SharedPreferenceProvider.java
package dev.skomlach.common.cryptostorage;

import android.content.SharedPreferences;

import dev.skomlach.common.contextprovider.AndroidContext;

public class SharedPreferenceProvider {

    private static CryptoPreferencesProvider dependencies;

    public static SharedPreferences getCryptoPreferences(String name) {
        if (dependencies == null) {
            dependencies = new EncryptedPreferencesProvider(AndroidContext.getAppContext());
        }
        return dependencies.getCryptoPreferences(name);
    }
}

File: common/src/main/java/dev/skomlach/common/cryptostorage/SharedPreferenceProvider.kt
package dev.skomlach.common.cryptostorage

import android.content.SharedPreferences
import dev.skomlach.common.contextprovider.AndroidContext.appContext

object SharedPreferenceProvider {
    private var dependencies: CryptoPreferencesProvider? = null

    @JvmStatic
    @Synchronized
    fun getCryptoPreferences(name: String): SharedPreferences {
        if (dependencies == null) {
            dependencies = EncryptedPreferencesProvider(appContext)
        }
        return (dependencies as CryptoPreferencesProvider).getCryptoPreferences(name)
    }
}